x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryRequest.java

```diff
@@ -84,11 +94,14 @@ public ActionRequestValidationException validate() {
         if (Strings.hasText(query) == false) {
             validationException = addValidationError("[" + RequestXContent.QUERY_FIELD + "] is required", validationException);
         }
-        if (onSnapshotBuild == false && pragmas.isEmpty() == false) {
-            validationException = addValidationError(
-                "[" + RequestXContent.PRAGMA_FIELD + "] only allowed in snapshot builds",
-                validationException
-            );
+        if (onSnapshotBuild == false) {
+            if (pragmas.isEmpty() == false) {
+                validationException = addValidationError(
+                    "[" + RequestXContent.PRAGMA_FIELD + "] only allowed in snapshot builds",
+                    validationException
+                );
+            }
```

Comment:
I think we should reject requests with tables in non-snapshot build for now?

Reply:
OH! I totally had that in there at some point. I'll add it back.